### PR TITLE
Fix dock pane layout

### LINKED ISSUE #442: Dock pane size control fails with Qt5
[This code](https://github.com/enthought/pyface/blob/e047d8b882f2d2f52b4a4a8e7c6af86be2489143/pyface/ui/qt4/tasks/main_window_layout.py#L270-L273) is supposed to allow control of the width and height of the dock panes in a tasks window.

It works as desired with PyQt 4, but apparently no longer works with PyQt 5.

To reproduce:
- build a PyQt 5-based test environment (e.g., with etstool, pyface-test-3.6-pyqt5)
- run `examples/tasks/basic/run.py` in that environment to show one of the example applications
- in `run.py`, add "width=500" to the `PaneItem` in `_default_layout_default`.
- run the application again. Expected behaviour is that the left-hand pane has width 500 pixels. Observed behaviour is that the value of `500` is ignored.

Under PyQt 4, the width is respected.

Digging deeper, PyQt4 uses a rather horrible hack to make this work: it sets both the maximum and minimum size of the dock pane's child using the given layout information (resulting in a fixed size pane), and then after Qt had performed layout, [removes those restrictions](https://github.com/enthought/pyface/blob/e047d8b882f2d2f52b4a4a8e7c6af86be2489143/pyface/ui/qt4/tasks/main_window_layout.py#L282-L295), resetting the max and min sizes of the dock pane and its child.

This hack apparently no longer works on PyQt5.

From reading, the right way to fix this may be to have a custom subclass of `QWidget` that supplies the appropriate size hint on demand.

### PR DESCRIPTION
**Pull request by pedrorivotti**
This PR is an alternative to #535 that also closes #442 

@rahulporuri Could you please check if this approach works for your use case?

### CODE REVIEW

**Comment by rahulporuri** [comment]
This PR definitely fixes the issue unlike #535 BUT understandably, this breaks behavior on pyqt 4 🤣

File: pyface/ui/qt4/tasks/main_window_layout.py [hunk 1]
```diff
@@ -154,9 +154,6 @@ def set_layout(self, layout):
                     sublayout, q_dock_area, _toplevel_call=False
                 )
 
-        # Remove the fixed sizes once Qt activates the layout.
-        QtCore.QTimer.singleShot(0, self._reset_fixed_sizes)
-
```
Comment (rahulporuri) [inline pyface/ui/qt4/tasks/main_window_layout.py:159]: can you undo the removals and add if/else clauses to not change the behavior on qt4? like in #535 . (i haven't checked the test suite but i'd wager reverting the changes would fix the test suite)

Not sure how to write a regression test for this though.

**Comment by rahulporuri** [comment]
@pedrorivotti will you be able to work on this PR this week? If not, I can take over and update the PR (unless someone on the ETS team can take over @corranwebster ). It'd be nice to get this merged and shipped in a bug fix soon.

**Comment by pedrorivotti** [comment]
@rahulporuri Please feel free to take over, I don't think I would get around to it before next week. 
Thanks!

File: pyface/ui/qt4/tasks/main_window_layout.py (hunk 1 above)
Comment (rahulporuri) [reply pyface/ui/qt4/tasks/main_window_layout.py]: i restored the removed code and moved into a qt4 conditional branch. but i'm still not sure how to add a regression test for this.

**Comment by rahulporuri** [comment]
@kitchoi @corranwebster any guidance on adding a regression test for this? I have tested this locally on our application with qt4 and qt5 - and dock pane restoration works as expected on both toolkits.

**Comment by kitchoi** [comment]
Confirmed with @corranwebster offline that this can be merged.

**Comment by rahulporuri** [comment]
@kitchoi are there plans on a bugfix release?

**Comment by kitchoi** [comment]
I just opened a bug release check list issue for TraitsUI, and will open one for Pyface. It is not really a complete plan as in there isn't a target release date yet. Open for discussion!

**Pull request merged**